examples/Makefile

```diff
@@ -1,25 +1,13 @@
+## this is a simple wrapper just to forward on known commands to the
+## embedding example with values pulled from Make.inc
 SRCDIR := $(abspath $(dir $(lastword $(MAKEFILE_LIST))))
-BUILDDIR := .
 JULIAHOME := $(abspath $(SRCDIR)/..)
+BUILDDIR := .
 include $(JULIAHOME)/Make.inc
 
-outdir := $(libexecdir)
-
-embedding_binary := $(abspath $(outdir)/embedding$(JULIA_LIBSUFFIX)$(EXE))
-
-release: embedding
-debug: embedding-debug
-
-embedding: $(embedding_binary)
-embedding-debug: $(embedding_binary)
-
-$(embedding_binary): $(wildcard embedding/*)
-	@$(MAKE) $(QUIET_MAKE) -C $(BUILDROOT)/examples/embedding $(JULIA_BUILD_MODE) \
-                                JULIA="$(bindir)/julia$(JULIA_LIBSUFFIX)$(EXE)" BIN="$(outdir)" \
-                                SPAWN="$(spawn)" CC="$(CC)"
-
-clean:
-	-rm -f $(embedding_binary) $(embedding_binary)-debug
-
-.PHONY: all embedding clean
-
+release: # default target
+# forward all variables expected by the embedding example
+JULIA:=$(call spawn,$(JULIA_EXECUTABLE))
+BIN:=$(build_bindir)
```

Comment:
1. Our makefile is a significant oddball in *not* providing a `make check` target. Nearly all of our dependencies do so.

2. Sure, but entirely irrelevant to this PR. That scenario is also completely broken with our current setup. At some point, I (we) will put time into fixing that. The current Makefile configuration is going to make that harder though, which is what triggered this PR to make it more sane at handling the embedding test in a simpler way.